### ISSUE TITLE
Unit test errors in github validation builds

### ISSUE THREAD

**Issue by marcushoepfner** [eclipse-platform/eclipse.platform.ui#665]
Refer to #662 

There are unit test failures in:
- **Windows** org.eclipse.jface.tests.notifications.NotificationPopupTest
```
Expected: a collection containing is <Text {}>
     but: was <List {}>
java.lang.AssertionError: 

Expected: a collection containing is <Text {}>
     but: was <List {}>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at org.eclipse.jface.tests.notifications.NotificationPopupTest.createsWithTextContent(NotificationPopupTest.java:100)
```
- **Windows** org.eclipse.ui.internal.monitoring.EventLoopMonitorThreadTests
```
A freeze event didn't capture a good range of stack samples (462) expected:<5> but was:<1>
java.lang.AssertionError: A freeze event didn't capture a good range of stack samples (462) expected:<5> but was:<1>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.eclipse.ui.internal.monitoring.EventLoopMonitorThreadTests.testEventLogging(EventLoopMonitorThreadTests.java:358)
```
- **Windows** org.eclipse.ui.internal.ide.application.FileLocationDetailsTest (several)
```
Illegal char <:> at index 87: C:\Users\RUNNER~1\AppData\Local\Temp\FileLocationDetailsTest3143935122662222144java+178:3
java.nio.file.InvalidPathException: Illegal char <:> at index 87: C:\Users\RUNNER~1\AppData\Local\Temp\FileLocationDetailsTest3143935122662222144java+178:3
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at org.eclipse.core.internal.filesystem.local.nio.DosHandler.fetchFileInfo(DosHandler.java:42)
	at org.eclipse.core.internal.filesystem.local.LocalFileNativesManager.fetchFileInfo(LocalFileNativesManager.java:65)
	at org.eclipse.core.internal.filesystem.local.LocalFile.fetchInfo(LocalFile.java:161)
	at org.eclipse.core.filesystem.provider.FileStore.fetchInfo(FileStore.java:260)
	at org.eclipse.ui.internal.ide.application.FileLocationDetails.checkLocation(FileLocationDetails.java:105)
	at org.eclipse.ui.internal.ide.application.FileLocationDetails.resolve(FileLocationDetails.java:63)
	at org.eclipse.ui.internal.ide.application.FileLocationDetailsTest.testLineAsPlus_ColumnAsColon(FileLocationDetailsTest.java:81)
```

They seem to be in the main branch.

**Comment by iloveeclipse** [comment]
@marcushoepfner : I've updated bug description to have OS and stack information.

**Comment by marcushoepfner** [comment]
Fixed the JST problem with #691 

Can someone using Win have a look at FileLocationDetailsTest. According to the call stack, it might even be related to the used JVM (Refer to initial comment.)? Or am I wrong? 

I wonder if there is an issue in the tests only a real functional bug.

Looking at at the assert `Illegal char <:> at index 87: C:\Users\RUNNER~1\AppData\Local\Temp\FileLocationDetailsTest3143935122662222144java+178:3` it at least seems that there is a real bug in the file positions.

**Comment by oleosterhagen** [comment]
`NotificationPopupTest` and `FileLocationDetailsTest` have been fixed with commit 23f20b814d347d35f6db87492fba793bfcacfa62 by @HannesWell. Thank you!

For the sporadically failing `EventLoopMonitorThreadTests` I have a solution as part of #669. This PR needs a reviewer / commiter.

**Comment by marcushoepfner** [comment]
Thanks a lot. This really helps.
